book/listings/actions/7/window/imp.rs

```diff
@@ -40,14 +40,14 @@ impl ObjectSubclass for Window {
 // ANCHOR: object_impl
 // Trait shared by all GObjects
 impl ObjectImpl for Window {
-    fn constructed(&self, obj: &Self::Type) {
+    fn constructed(&self) {
         // Call "constructed" on parent
-        self.parent_constructed(obj);
+        self.parent_constructed();
 
         // Setup
-        obj.setup_settings();
-        obj.setup_actions();
-        obj.bind_settings();
+        self.instance().setup_settings();
```

Comment:
I would suggest calling self.instance() only once and store it in an obj / window variable.

Reply:
I don't think there should be any, cc @sdroege for confirming that. But it doesn't make much sense to repeat the same call if you can store the obj once and re-use it.